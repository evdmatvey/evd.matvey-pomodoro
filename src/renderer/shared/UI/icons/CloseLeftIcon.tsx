import { FC } from 'react';
import type { IconProps } from './IconProps';

const CloseLeftIcon: FC<IconProps> = ({ width = 30, height = 30 }) => {
  return (
    <svg
      width={width}
      height={height}
      viewBox="0 0 30 30"
      fill="none"
      xmlns="http://www.w3.org/2000/svg">
      <path
        d="M26.25 13.75H11.7625L14.6375 10.8875C14.754 10.771 14.8465 10.6326 14.9096 10.4803C14.9727 10.328 15.0051 10.1648 15.0051 10C15.0051 9.83518 14.9727 9.67197 14.9096 9.51969C14.8465 9.36741 14.754 9.22905 14.6375 9.1125C14.521 8.99595 14.3826 8.9035 14.2303 8.84043C14.078 8.77735 13.9148 8.74489 13.75 8.74489C13.5852 8.74489 13.422 8.77735 13.2697 8.84043C13.1174 8.9035 12.979 8.99595 12.8625 9.1125L7.8625 14.1125C7.7487 14.2314 7.65949 14.3716 7.6 14.525C7.47498 14.8293 7.47498 15.1707 7.6 15.475C7.65949 15.6284 7.7487 15.7686 7.8625 15.8875L12.8625 20.8875C12.9787 21.0047 13.117 21.0977 13.2693 21.1611C13.4216 21.2246 13.585 21.2572 13.75 21.2572C13.915 21.2572 14.0784 21.2246 14.2307 21.1611C14.383 21.0977 14.5213 21.0047 14.6375 20.8875C14.7547 20.7713 14.8477 20.633 14.9111 20.4807C14.9746 20.3284 15.0072 20.165 15.0072 20C15.0072 19.835 14.9746 19.6716 14.9111 19.5193C14.8477 19.367 14.7547 19.2287 14.6375 19.1125L11.7625 16.25H26.25C26.5815 16.25 26.8995 16.1183 27.1339 15.8839C27.3683 15.6495 27.5 15.3315 27.5 15C27.5 14.6685 27.3683 14.3505 27.1339 14.1161C26.8995 13.8817 26.5815 13.75 26.25 13.75ZM3.75 3.75C3.41848 3.75 3.10054 3.8817 2.86612 4.11612C2.6317 4.35054 2.5 4.66848 2.5 5V25C2.5 25.3315 2.6317 25.6495 2.86612 25.8839C3.10054 26.1183 3.41848 26.25 3.75 26.25C4.08152 26.25 4.39946 26.1183 4.63388 25.8839C4.8683 25.6495 5 25.3315 5 25V5C5 4.66848 4.8683 4.35054 4.63388 4.11612C4.39946 3.8817 4.08152 3.75 3.75 3.75V3.75Z"
        fill="#ADADAD"
      />
    </svg>
  );
};

export default CloseLeftIcon;
