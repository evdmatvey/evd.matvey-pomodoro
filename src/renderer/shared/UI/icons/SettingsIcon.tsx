import { FC } from 'react';
import type { IconProps } from './IconProps';

const SettingsIcon: FC<IconProps> = ({ width = 30, height = 30 }) => {
  return (
    <svg
      width={width}
      height={height}
      viewBox="0 0 30 30"
      fill="none"
      xmlns="http://www.w3.org/2000/svg">
      <path
        d="M26.65 11.9375L24.2875 11.15L25.4 8.925C25.5128 8.69211 25.5505 8.42996 25.508 8.17472C25.4654 7.91948 25.3447 7.68373 25.1625 7.5L22.5 4.8375C22.3153 4.65261 22.0773 4.53026 21.8194 4.48766C21.5616 4.44506 21.2969 4.48435 21.0625 4.6L18.8375 5.7125L18.05 3.35C17.9668 3.10374 17.809 2.88953 17.5984 2.73719C17.3878 2.58484 17.1349 2.50193 16.875 2.5H13.125C12.8629 2.49932 12.6073 2.58102 12.3942 2.73355C12.1811 2.88608 12.0214 3.10172 11.9375 3.35L11.15 5.7125L8.925 4.6C8.69211 4.48723 8.42996 4.44951 8.17472 4.49205C7.91948 4.53459 7.68373 4.65529 7.5 4.8375L4.8375 7.5C4.65261 7.68471 4.53026 7.92272 4.48766 8.18057C4.44506 8.43842 4.48435 8.70314 4.6 8.9375L5.7125 11.1625L3.35 11.95C3.10374 12.0332 2.88954 12.191 2.73719 12.4016C2.58484 12.6122 2.50193 12.8651 2.5 13.125V16.875C2.49933 17.1371 2.58103 17.3927 2.73356 17.6058C2.88609 17.8189 3.10173 17.9786 3.35 18.0625L5.7125 18.85L4.6 21.075C4.48723 21.3079 4.44951 21.57 4.49205 21.8253C4.53459 22.0805 4.65529 22.3163 4.8375 22.5L7.5 25.1625C7.68471 25.3474 7.92272 25.4697 8.18057 25.5123C8.43842 25.5549 8.70314 25.5157 8.9375 25.4L11.1625 24.2875L11.95 26.65C12.0339 26.8983 12.1936 27.1139 12.4067 27.2664C12.6198 27.419 12.8754 27.5007 13.1375 27.5H16.8875C17.1496 27.5007 17.4052 27.419 17.6183 27.2664C17.8314 27.1139 17.9911 26.8983 18.075 26.65L18.8625 24.2875L21.0875 25.4C21.3189 25.5099 21.5785 25.5461 21.8311 25.5036C22.0838 25.4611 22.3173 25.3421 22.5 25.1625L25.1625 22.5C25.3474 22.3153 25.4697 22.0773 25.5123 21.8194C25.5549 21.5616 25.5157 21.2969 25.4 21.0625L24.2875 18.8375L26.65 18.05C26.8963 17.9668 27.1105 17.809 27.2628 17.5984C27.4152 17.3878 27.4981 17.1349 27.5 16.875V13.125C27.5007 12.8629 27.419 12.6073 27.2665 12.3942C27.1139 12.1811 26.8983 12.0214 26.65 11.9375ZM25 15.975L23.5 16.475C23.1551 16.5869 22.8386 16.7725 22.5726 17.0189C22.3065 17.2653 22.0973 17.5666 21.9593 17.902C21.8214 18.2373 21.758 18.5987 21.7737 18.961C21.7894 19.3233 21.8836 19.6778 22.05 20L22.7625 21.425L21.3875 22.8L20 22.05C19.6794 21.8903 19.3283 21.8013 18.9704 21.7891C18.6125 21.7769 18.2561 21.8418 17.9254 21.9792C17.5947 22.1167 17.2974 22.3237 17.0537 22.586C16.8099 22.8484 16.6253 23.1601 16.5125 23.5L16.0125 25H14.025L13.525 23.5C13.4131 23.1551 13.2275 22.8386 12.9811 22.5726C12.7347 22.3065 12.4334 22.0973 12.098 21.9593C11.7627 21.8214 11.4013 21.758 11.039 21.7737C10.6768 21.7894 10.3222 21.8836 10 22.05L8.575 22.7625L7.2 21.3875L7.95 20C8.11637 19.6778 8.21065 19.3233 8.22631 18.961C8.24197 18.5987 8.17863 18.2373 8.04068 17.902C7.90273 17.5666 7.69348 17.2653 7.42744 17.0189C7.16139 16.7725 6.84493 16.5869 6.5 16.475L5 15.975V14.025L6.5 13.525C6.84493 13.4131 7.16139 13.2275 7.42744 12.9811C7.69348 12.7347 7.90273 12.4334 8.04068 12.098C8.17863 11.7627 8.24197 11.4013 8.22631 11.039C8.21065 10.6767 8.11637 10.3222 7.95 10L7.2375 8.6125L8.6125 7.2375L10 7.95C10.3222 8.11637 10.6768 8.21065 11.039 8.22631C11.4013 8.24196 11.7627 8.17862 12.098 8.04067C12.4334 7.90272 12.7347 7.69347 12.9811 7.42743C13.2275 7.16139 13.4131 6.84493 13.525 6.5L14.025 5H15.975L16.475 6.5C16.5869 6.84493 16.7725 7.16139 17.0189 7.42743C17.2653 7.69347 17.5666 7.90272 17.902 8.04067C18.2373 8.17862 18.5987 8.24196 18.961 8.22631C19.3233 8.21065 19.6778 8.11637 20 7.95L21.425 7.2375L22.8 8.6125L22.05 10C21.8903 10.3206 21.8013 10.6717 21.7891 11.0296C21.7769 11.3875 21.8418 11.7439 21.9792 12.0746C22.1167 12.4053 22.3237 12.7026 22.5861 12.9464C22.8484 13.1901 23.1601 13.3747 23.5 13.4875L25 13.9875V15.975ZM15 10C14.0111 10 13.0444 10.2932 12.2222 10.8427C11.3999 11.3921 10.759 12.173 10.3806 13.0866C10.0022 14.0002 9.90315 15.0055 10.0961 15.9755C10.289 16.9454 10.7652 17.8363 11.4645 18.5355C12.1637 19.2348 13.0546 19.711 14.0246 19.9039C14.9945 20.0969 15.9998 19.9978 16.9134 19.6194C17.8271 19.241 18.6079 18.6001 19.1574 17.7779C19.7068 16.9556 20 15.9889 20 15C20 13.6739 19.4732 12.4021 18.5355 11.4645C17.5979 10.5268 16.3261 10 15 10ZM15 17.5C14.5056 17.5 14.0222 17.3534 13.6111 17.0787C13.2 16.804 12.8795 16.4135 12.6903 15.9567C12.5011 15.4999 12.4516 14.9972 12.548 14.5123C12.6445 14.0273 12.8826 13.5819 13.2322 13.2322C13.5819 12.8826 14.0273 12.6445 14.5123 12.548C14.9972 12.4516 15.4999 12.5011 15.9567 12.6903C16.4135 12.8795 16.804 13.2 17.0787 13.6111C17.3534 14.0222 17.5 14.5055 17.5 15C17.5 15.663 17.2366 16.2989 16.7678 16.7678C16.2989 17.2366 15.663 17.5 15 17.5Z"
        fill="#ADADAD"
      />
    </svg>
  );
};

export default SettingsIcon;
