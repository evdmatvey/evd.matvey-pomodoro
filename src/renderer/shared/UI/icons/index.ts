import SettingsIcon from './SettingsIcon';
import CloseLeftIcon from './CloseLeftIcon';
import PauseIcon from './PauseIcon';

export { SettingsIcon, CloseLeftIcon, PauseIcon };
