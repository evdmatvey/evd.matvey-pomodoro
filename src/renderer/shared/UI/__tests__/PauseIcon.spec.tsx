import { test, expect } from 'vitest';
import { render } from '@testing-library/react';
import { PauseIcon } from '../icons';

test('PauseIcon snapshot', () => {
  const pauseIcon = render(<PauseIcon />);
  expect(pauseIcon).toMatchSnapshot();
});
