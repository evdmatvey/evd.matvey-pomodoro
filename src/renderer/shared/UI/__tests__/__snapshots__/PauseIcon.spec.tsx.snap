// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PauseIcon snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        fill="none"
        viewBox="0 0 30 30"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20 2.5C19.0054 2.5 18.0516 2.89509 17.3483 3.59835C16.6451 4.30161 16.25 5.25544 16.25 6.25V23.75C16.25 24.7446 16.6451 25.6984 17.3483 26.4017C18.0516 27.1049 19.0054 27.5 20 27.5C20.9946 27.5 21.9484 27.1049 22.6517 26.4017C23.3549 25.6984 23.75 24.7446 23.75 23.75V6.25C23.75 5.25544 23.3549 4.30161 22.6517 3.59835C21.9484 2.89509 20.9946 2.5 20 2.5ZM21.25 23.75C21.25 24.0815 21.1183 24.3995 20.8839 24.6339C20.6495 24.8683 20.3315 25 20 25C19.6685 25 19.3505 24.8683 19.1161 24.6339C18.8817 24.3995 18.75 24.0815 18.75 23.75V6.25C18.75 5.91848 18.8817 5.60054 19.1161 5.36612C19.3505 5.1317 19.6685 5 20 5C20.3315 5 20.6495 5.1317 20.8839 5.36612C21.1183 5.60054 21.25 5.91848 21.25 6.25V23.75ZM10 2.5C9.00544 2.5 8.05161 2.89509 7.34835 3.59835C6.64509 4.30161 6.25 5.25544 6.25 6.25V23.75C6.25 24.7446 6.64509 25.6984 7.34835 26.4017C8.05161 27.1049 9.00544 27.5 10 27.5C10.9946 27.5 11.9484 27.1049 12.6517 26.4017C13.3549 25.6984 13.75 24.7446 13.75 23.75V6.25C13.75 5.25544 13.3549 4.30161 12.6517 3.59835C11.9484 2.89509 10.9946 2.5 10 2.5ZM11.25 23.75C11.25 24.0815 11.1183 24.3995 10.8839 24.6339C10.6495 24.8683 10.3315 25 10 25C9.66848 25 9.35054 24.8683 9.11612 24.6339C8.8817 24.3995 8.75 24.0815 8.75 23.75V6.25C8.75 5.91848 8.8817 5.60054 9.11612 5.36612C9.35054 5.1317 9.66848 5 10 5C10.3315 5 10.6495 5.1317 10.8839 5.36612C11.1183 5.60054 11.25 5.91848 11.25 6.25V23.75Z"
          fill="#ADADAD"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      fill="none"
      viewBox="0 0 30 30"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20 2.5C19.0054 2.5 18.0516 2.89509 17.3483 3.59835C16.6451 4.30161 16.25 5.25544 16.25 6.25V23.75C16.25 24.7446 16.6451 25.6984 17.3483 26.4017C18.0516 27.1049 19.0054 27.5 20 27.5C20.9946 27.5 21.9484 27.1049 22.6517 26.4017C23.3549 25.6984 23.75 24.7446 23.75 23.75V6.25C23.75 5.25544 23.3549 4.30161 22.6517 3.59835C21.9484 2.89509 20.9946 2.5 20 2.5ZM21.25 23.75C21.25 24.0815 21.1183 24.3995 20.8839 24.6339C20.6495 24.8683 20.3315 25 20 25C19.6685 25 19.3505 24.8683 19.1161 24.6339C18.8817 24.3995 18.75 24.0815 18.75 23.75V6.25C18.75 5.91848 18.8817 5.60054 19.1161 5.36612C19.3505 5.1317 19.6685 5 20 5C20.3315 5 20.6495 5.1317 20.8839 5.36612C21.1183 5.60054 21.25 5.91848 21.25 6.25V23.75ZM10 2.5C9.00544 2.5 8.05161 2.89509 7.34835 3.59835C6.64509 4.30161 6.25 5.25544 6.25 6.25V23.75C6.25 24.7446 6.64509 25.6984 7.34835 26.4017C8.05161 27.1049 9.00544 27.5 10 27.5C10.9946 27.5 11.9484 27.1049 12.6517 26.4017C13.3549 25.6984 13.75 24.7446 13.75 23.75V6.25C13.75 5.25544 13.3549 4.30161 12.6517 3.59835C11.9484 2.89509 10.9946 2.5 10 2.5ZM11.25 23.75C11.25 24.0815 11.1183 24.3995 10.8839 24.6339C10.6495 24.8683 10.3315 25 10 25C9.66848 25 9.35054 24.8683 9.11612 24.6339C8.8817 24.3995 8.75 24.0815 8.75 23.75V6.25C8.75 5.91848 8.8817 5.60054 9.11612 5.36612C9.35054 5.1317 9.66848 5 10 5C10.3315 5 10.6495 5.1317 10.8839 5.36612C11.1183 5.60054 11.25 5.91848 11.25 6.25V23.75Z"
        fill="#ADADAD"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
